### ISSUE TITLE
How can I study a model during training?

### ISSUE THREAD

**Issue by joemrt** [sdv-dev/SDV#1167]
I would like to study the convergence speed of a CopulaGAN. To do so, I would like to either 
- Interrupt the training, analyze the model and resume the training or
- Save intermediate checkpoints of the model

Is at least one of these options possible at the moment? 

Please apologize if the information is somewhere in the documentation and I didn't see it.

**Comment by npatki** [comment]
Hi @joemrt, nice to meet you! 

Unfortunately, this isn't possible at the moment. We are tracking a closely related feature request to train the model in batches in #805. I think this will make your use case possible too. You're welcome to follow along with this issue and add any info to help us prioritize.

Until then, you can get some learnings by creating multiple models and run them for different epochs. I recognize this is a resource-intensive approach, but setting a seed should at least ensure they are training in the same way.

```python
np.random.seed(0)
torch.manual_seed(0)
model_10_epochs = CopulaGAN(epochs=10)
model_10_epochs.fit(data)

np.random.seed(0)
torch.manual_seed(0)
model_100_epochs = CopulaGAN(epochs=100)
model_100_epochs.fit(data)

etc.
```

**Comment by joemrt** [comment]
Hi @npatki,
The "for each checkpoint a seperate training" approach is indeed computationally heavy, but should be manageable for my purpose. Thanks!
The saving of intermediate checkpoints or ability to resume the training could be a really useful option in my eyes if you think about extending the framework at some point. Thanks again a lot for your help!

**Comment by npatki** [comment]
Glad we could help @joemrt! I'll close the issue as resolved since we're using #805 to track the overall request. Feel free to reply if you have any other thoughts.
